src/util/dsp/yule_walker.cpp

```diff
@@ -0,0 +1,98 @@
+//**************************************************************************
+//
+//	$file: yule_walker.c $
+//	$author: Martin Fouilleul $
+//	$date: 28/11/2018 $
+//	$description: a simple Yule-Walker system solver$
+//
+//**************************************************************************
+
+#include<string.h> // memset()
+
+namespace otto::dsp
+{
+//--------------------------------------------------------------------------
+//
+//	SolveYuleWalker() is a simple solver for Yule-Walker equations using a specialized version of the Levinson-Durbin algorithm.
+//
+//	arguments :
+//		- size : the size of the Toeplitz matrix and of result vector sigmaAndCoeffs
+//		- acov : a buffer containing the covariance coefficents for indices 0 to (size-1).
+//			     ie. [acov(0) acov(1) ... acov(n-1)] of length (size)
+//
+//		- sigmaAndCoeffs : result vector of length (size). The first element will be the variance of the signal, and the remaining elements will be the estimated coefficients of the autoregressive filter.
+//		- scratchBuffer : a block of memory used to hold intermediate computations. You must pass a buffer of size at least (2 * size * sizeof(float)).
+//
+//--------------------------------------------------------------------------
+
+void SolveYuleWalker(unsigned int size, const float* acov, float* sigmaAndCoeffs, char* scratchBuffer)
```

Comment:
```suggestion
void solve_yule_walker(gsl::span<const float> acov, gsl::span<float> sigma_and_coeffs, gsl::span<float> scratch_buffer)
```
Then document the size requirements.

using span and not pointers is not for the bounds checking, its for the better interface.